tests/configlet/util/base_test.py

```diff
@@ -0,0 +1,301 @@
+
+    append_log_prefix_msg("load_mini_wo_t0", pfx_lvl)
+    load_minigraph(duthost)
+
+    append_log_prefix_msg("load_mini_wo_t0", pfx_lvl)
+    take_DB_dumps(duthost, no_t0_db_dir, data_dir)
+
+    set_prefix_lvl(pfx_lvl)
+
+
+def apply_clet(duthost, skip_test=False):
+
+    pfx_lvl = get_prefix_lvl()
+
+    clet_file = managed_files["configlet"]
+    sonic_clet_file = "/etc/sonic/add_Rack.json"
+
+    del_clet_file = os.path.join(os.path.dirname(
+        os.path.abspath(__file__)), "everflow_delete.json")
+    del_sonic_clet_file = "/etc/sonic/everflow_delete.json"
+
+    if not clet_file:
+        report_error("Failed to get files wo_to={} clet={}".format(
+            mini_wo_to, clet_file))
+
+    if not os.path.exists(clet_file):
+        report_error("configlet {} file absent".format(clet_file))
+
+    duthost.copy(src=clet_file, dest=sonic_clet_file)
+    duthost.copy(src=del_clet_file, dest=del_sonic_clet_file)
+
+    append_log_prefix_msg("applying", pfx_lvl)
+    # Apply delete 
+    duthost.shell("configlet -d -j {}".format(del_sonic_clet_file))
+
+    tor_ifname = tor_data["links"][0]["local"]["sonic_name"]
+    duthost.shell("config interface shutdown {}".format(tor_ifname))
+    do_pause(PAUSE_INTF_DOWN, "pause upon i/f {} shutdown".format(tor_ifname))
+
+    duthost.shell("configlet -u -j {}".format(sonic_clet_file))
+    do_pause(PAUSE_CLET_APPLY, "Pause after applying configlet")
+
+    duthost.shell("config interface startup {}".format(tor_ifname))
+    do_pause(PAUSE_INTF_UP, "pause upon i/f {} startup".format(tor_ifname))
+
+    append_log_prefix_msg("checking_dump", pfx_lvl)
+    assert wait_until(DB_COMP_WAIT_TIME, 20, 0, db_comp, duthost, clet_db_dir,
+            orig_db_dir, "apply_clet"), \
+            "DB compare failed after apply-clet"
+
+    # Ensure BGP session is up
+    chk_bgp_session(duthost, tor_data["ip"]["remote"], "post-clet test")
+    chk_bgp_session(duthost, tor_data["ipv6"]["remote"].lower(), "post-clet test")
+
+    log_info("AddRack by template succeeded")
+
+    set_prefix_lvl(pfx_lvl)
+
+
+def download_sonic_files(duthost):
+    for f in [ "minigraph.xml", "config_db.json" ]:
+        duthost.fetch(src="/etc/sonic/{}".format(f), dest=data_dir)
+
+def files_create(is_mlnx, is_storage_backend):
+    init_global_data()
+
+    strip.main(init_data["files_dir"])
+    configlet.main(init_data["files_dir"], is_mlnx, is_storage_backend)
+
+    log_info("Managed files: {}".format(json.dumps(managed_files, indent=4)))
+                                
+                                        
+def do_test_add_rack(duthost, is_storage_backend = False, skip_load=False,
+        skip_clet_test=False, skip_generic_add=False, skip_generic_rm=False,
+        hack_apply=False, skip_prepare = False):
+
+    global data_dir, orig_db_dir, clet_db_dir, files_dir
+
+    init(duthost)
+
+    fpath = "/usr/local/lib/python3.9/dist-packages/generic_config_updater/services_validator.py"
```

Comment:
>/usr/local/lib/python3.9/dist-packages/generic_config_updater/services_validator.py [](http://example.com/codeflow?start=13&length=83)

Python version keeps increasing along with Debian version. So it may not future proof in test.

If your purpose is to check feature availability. You may use `python3 -c 'import generic_config_updater'`.

Reply:
Good point. Resolved.